lib/core/hook.js

```diff
@@ -0,0 +1,35 @@
+import { routeOption, getMatchedComponents, normalizePath } from './utilities'
```

Comment:
- Cloning a Nuxt template into the module is not probably the best idea
- I would suggest creating a new issue first discussing on hook possibility and sperate from this PR
- A full rewrite is planned of auth module to use hook functions like this :)

Reply:
Now that I'm looking back at this, I think this PR has changed from a really old change I was gonna propose, with this hook. Which sucks cuz I spent the time building this change off of that :(

This PR will require a rework then. I think the same concepts will apply, but I have to reset to master before doing them

Reply:
These should be resolved now @pi0

Reply:
Removed, part of a different branch incorrectly included.